frontend/packages/ux-editor/src/components/config/editModal/EditOptions/EditCodeList/OptionListEditor.test.tsx

```diff
@@ -0,0 +1,171 @@
+
+// Test data:
+const mockComponent: FormComponent<ComponentType.Dropdown> = componentMocks[ComponentType.Dropdown];
+mockComponent.optionsId = 'options';
+
+const apiResult = {
+  options: [
+    { value: 'test', label: 'label text', description: 'description', helpText: 'help text' },
+    { value: 2, label: 'label number', description: null, helpText: null },
+    { value: true, label: 'label boolean', description: null, helpText: null },
+  ],
+};
+
+const queryClientMock = createQueryClientMock();
+
+describe('OptionListEditor', () => {
+  afterEach(() => {
+    queryClientMock.clear();
+  });
+
+  it('should render a spinner when there is no data', () => {
+    renderOptionListEditor({
+      queries: {
+        getOptionLists: jest
+          .fn()
+          .mockImplementation(() => Promise.resolve<Record<string, Option[]>>({})),
+      },
+    });
+
+    expect(
+      screen.getByText(textMock('ux_editor.modal_properties_code_list_spinner_title')),
+    ).toBeInTheDocument();
+  });
+
+  it('should render a error message when api throws an error', async () => {
+    renderOptionListEditor({
+      queries: {
+        getOptionLists: jest.fn().mockRejectedValueOnce(new Error('Error')),
+      },
+    });
+    await waitForElementToBeRemoved(() => {
+      return screen.queryByText(textMock('ux_editor.modal_properties_code_list_spinner_title'));
+    });
+
+    expect(
+      screen.getByText(textMock('ux_editor.modal_properties_error_message')),
+    ).toBeInTheDocument();
+  });
+
+  it('should render the open Dialog button', async () => {
+    renderOptionListEditor();
+    await waitForElementToBeRemoved(() => {
+      return screen.queryByText(textMock('ux_editor.modal_properties_code_list_spinner_title'));
+    });
+
+    const btnOpen = screen.getByRole('button', {
+      name: textMock('ux_editor.modal_properties_code_list_open_editor'),
+    });
+
+    expect(btnOpen).toBeInTheDocument();
+  });
+
+  it('should open Dialog', async () => {
+    const user = userEvent.setup();
+    renderOptionListEditor();
+    await waitForElementToBeRemoved(() => {
+      return screen.queryByText(textMock('ux_editor.modal_properties_code_list_spinner_title'));
+    });
+
+    await openModal(user);
+
+    expect(screen.getByRole('dialog')).toBeInTheDocument();
+  });
+
+  it('should close Dialog', async () => {
+    const user = userEvent.setup();
+    renderOptionListEditor();
+    await waitForElementToBeRemoved(() => {
+      return screen.queryByText(textMock('ux_editor.modal_properties_code_list_spinner_title'));
+    });
+
+    await openModal(user);
+
+    await user.click(screen.getByRole('button', { name: 'close modal' })); // Todo: Replace "close modal" with defaultDialogProps.closeButtonTitle when https://github.com/digdir/designsystemet/issues/2195 is fixed
+
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+  });
+
+  it('should call handleClose when closing Dialog', async () => {
```

Comment:
Is this test case title correct? Looks like we are testing if the preview reloads 🤔

Reply:
It tests if the preview reloads, which can only be done through handleClose. I'm not sure if the title. I can rename it to `should call doReloadPreview when closing Dialog` if that's more correct.

Reply:
That is better!